lib/ecto_ldap/adapter.ex

```diff
@@ -105,6 +156,9 @@ defmodule Ecto.Ldap.Adapter do
     {:nocache, query_metadata}
   end
 
+  def prepare(:update_all, _query), do: raise Exception, "Update is currently unsupported"
+  def prepare(:delete_all, _query), do: raise Exception, "Delete is currently unsupported"
```

Comment:
As per `dialyzer`:

```
adapter.ex:204: The call 'Elixir.UndefinedFunctionError':exception(<<_:248>>) 
will never return since the success typing is (maybe_improper_list()) -> #{} 
and the contract is ('Elixir.Keyword':t()) -> 'Elixir.Exception':t()
```